tracer/src/Datadog.Trace/ClrProfiler/TracerDebugger.cs

```diff
@@ -0,0 +1,57 @@
+// <copyright file="TracerDebugger.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Vendors.Serilog;
+
+namespace Datadog.Trace.ClrProfiler;
+
+// Based on: https://github.com/microsoft/vstest/blob/main/src/Microsoft.TestPlatform.Execution.Shared/DebuggerBreakpoint.cs#L25
+internal static class TracerDebugger
+{
+    // Based on: https://github.com/microsoft/vstest/blob/main/src/Microsoft.TestPlatform.Execution.Shared/DebuggerBreakpoint.cs#L140
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    internal static void WaitForDebugger(string environmentVariable)
```

Comment:
NITPICK: I would suggest to simplify it `WaitForDebugger()` and use `ConfigurationKeys.DebuggerEnabled` internally to avoid confusion / misuse.

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/3613/commits/2e5c371193338f4585b8ff83821cc2e9edcde605